### ISSUE TITLE
UnboundLocalError exception in `nikola deploy`

### ISSUE THREAD

**Issue by hiway** [getnikola/nikola#3479]
### Environment

**Python Version:**

Python 3.7.9

**Nikola Version:**

Nikola v8.1.2 (installed from git master)

**Operating System:** 

FreeBSD 12.2-RELEASE amd64

### Description:

On running `nikola deploy` on a new site, after rsync completes, this exception is thrown:

```python
  File "[...]/nikola/nikola/plugins/command/deploy.py", line 99, in _execute
    self._emit_deploy_event(last_deploy, new_deploy, clean, undeployed_posts)
UnboundLocalError: local variable 'clean' referenced before assignment
```

I looked at the source and noticed that `clean` is only set inside a condition:

https://github.com/getnikola/nikola/blob/d0ef5b68d0667f635ecb2bc0d5b92e480b5e4b25/nikola/plugins/command/deploy.py#L50-L56

To try and fix it, I changed the source to:

```python
        if last_deploy is not None:
            last_deploy = dateutil.parser.parse(last_deploy)
            clean = False
        else:
            clean = True
```

That leads to another exception elsewhere:

```python
  File "[...]/nikola/nikola/plugins/command/deploy.py", line 131, in _emit_deploy_event
    if last_deploy.tzinfo is None:
AttributeError: 'NoneType' object has no attribute 'tzinfo'
```

Which was fixed by changing the source code at:

https://github.com/getnikola/nikola/blob/d0ef5b68d0667f635ecb2bc0d5b92e480b5e4b25/nikola/plugins/command/deploy.py#L98-L99

to:

```python
        new_deploy = datetime.utcnow()
        if last_deploy is None:
            last_deploy = new_deploy
        self._emit_deploy_event(last_deploy, new_deploy, clean, undeployed_posts)
```

After this the deploy command has worked as I expected.

**Comment by dvzrv** [comment]
It would be awesome to see this fix in a bugfix release :)

**Comment by kayhayen** [comment]
If I am not mistaken, there is no such bug fix release, does anyone happen to know which was the last release that didn't have this regression? I would downgrade to that.

**Comment by Kwpolska** [comment]
I just released Nikola v8.1.3 with the fix for this.
